test/test_art_iter.cpp

```diff
@@ -0,0 +1,596 @@
+      UNODB_EXPECT_TRUE(it != e);
+      UNODB_EXPECT_EQ(match, true);
+      UNODB_EXPECT_TRUE(it.get_key() && it.get_key().value() == k0);
+      UNODB_EXPECT_TRUE(it.get_val() &&
+                        it.get_val().value() == unodb::test::test_values[0]);
+    }
+    {  // exact match, forward traversal (GTE), middle key.
+      bool match = false;
+      auto it = end(db).seek(unodb::detail::art_key{k1}, match, true /*fwd*/);
+      UNODB_EXPECT_TRUE(it != e);
+      UNODB_EXPECT_EQ(match, true);
+      UNODB_EXPECT_TRUE(it.get_key() && it.get_key().value() == k1);
+      UNODB_EXPECT_TRUE(it.get_val() &&
+                        it.get_val().value() == unodb::test::test_values[1]);
+    }
+    {  // exact match, forward traversal (GTE), last key.
+      bool match = false;
+      auto it = end(db).seek(unodb::detail::art_key{k2}, match, true /*fwd*/);
+      UNODB_EXPECT_TRUE(it != e);
+      UNODB_EXPECT_EQ(match, true);
+      UNODB_EXPECT_TRUE(it.get_key() && it.get_key().value() == k2);
+      UNODB_EXPECT_TRUE(it.get_val() &&
+                        it.get_val().value() == unodb::test::test_values[2]);
+    }
+  }
+  if (true) {  // exact match, reverse traversal
+    {          // exact match, reverse traversal (LTE), first key.
+      bool match = false;
+      auto it = end(db).seek(unodb::detail::art_key{k0}, match, false /*fwd*/);
+      UNODB_EXPECT_TRUE(it != e);
+      UNODB_EXPECT_EQ(match, true);
+      UNODB_EXPECT_TRUE(it.get_key() && it.get_key().value() == k0);
+      UNODB_EXPECT_TRUE(it.get_val() &&
+                        it.get_val().value() == unodb::test::test_values[0]);
+    }
+    {  // exact match, reverse traversal (LTE), middle key.
+      bool match = false;
+      auto it = end(db).seek(unodb::detail::art_key{k1}, match, false /*fwd*/);
+      UNODB_EXPECT_TRUE(it != e);
+      UNODB_EXPECT_EQ(match, true);
+      UNODB_EXPECT_TRUE(it.get_key() && it.get_key().value() == k1);
+      UNODB_EXPECT_TRUE(it.get_val() &&
+                        it.get_val().value() == unodb::test::test_values[1]);
+    }
+    {  // exact match, reverse traversal (LTE), last key.
+      bool match = false;
+      auto it = end(db).seek(unodb::detail::art_key{k2}, match, false /*fwd*/);
+      UNODB_EXPECT_TRUE(it != e);
+      UNODB_EXPECT_EQ(match, true);
+      UNODB_EXPECT_TRUE(it.get_key() && it.get_key().value() == k2);
+      UNODB_EXPECT_TRUE(it.get_val() &&
+                        it.get_val().value() == unodb::test::test_values[2]);
+    }
+  }
+  {              // before and after the first and last key
+    if (true) {  // forward traversal, before the first key in the data.
+      // match=false and iterator is positioned on the first key in the
+      // data.
+      bool match = true;
+      auto it = end(db).seek(unodb::detail::art_key{0}, match, true /*fwd*/);
+      if (debug) {
+        std::cerr << "end(db).seek(0,&match,true)::\n";
```

Comment:
The tests should be silent by default?

Reply:
it is. debug is a constexpr and is false

Reply:
Right, missed it